packages/shared/src/components/notifications/EnableNotification.tsx

```diff
@@ -3,48 +3,107 @@ import usePersistentContext from '../../hooks/usePersistentContext';
 import { Button } from '../buttons/Button';
 import NotificationsContext from '../../contexts/NotificationsContext';
 import { cloudinary } from '../../lib/image';
+import CloseButton from '../CloseButton';
+
+export enum NotificationPromptSource {
+  NotificationList = 'notificationList',
+  NewComment = 'newComment',
+  NewSource = 'newSource',
+}
 
 const DISMISS_BROWSER_PERMISSION = 'dismissBrowserPermission';
 
-function EnableNotification(): ReactElement {
-  const { hasPermission, requestPermission } = useContext(NotificationsContext);
-  const [dismissed, setDismissed, isLoaded] = usePersistentContext(
-    DISMISS_BROWSER_PERMISSION,
-    false,
-  );
+type DismissBrowserPermissions = {
+  [NotificationPromptSource.NotificationList]: boolean;
+  [NotificationPromptSource.NewComment]: boolean;
+  [NotificationPromptSource.NewSource]: boolean;
+};
 
+const DEFAULT_CACHE_VALUE: DismissBrowserPermissions = {
+  newComment: false,
+  newSource: false,
+  notificationList: false,
+};
```

Comment:
My only concern with this is that if we decide to introduce new places the cache will already be set.
Not urgently needed, we can decide to tackle if that happens.

Reply:
Yeah but that shouldn't have any downsides for us since we only care if the property has a truthy value. If the new property doesn't exist, it will have a falsy value by default.

```
const dismissed = !!dismissedCache?.[source];
```